plugin/pkg/scheduler/factory/factory.go

```diff
@@ -161,7 +163,82 @@ func NewConfigFactory(
 	)
 	c.nodeLister = nodeInformer.Lister()
 
-	// TODO(harryz) need to fill all the handlers here and below for equivalence cache
+	// On add and delete of PVs, it will affect equivalence cache items
+	// related to persistent volume
+	pvInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			// MaxPDVolumeCountPredicate: since it relies on the counts of PV.
+			AddFunc: func(obj interface{}) {
+				invalidPredicates := sets.NewString("MaxPDVolumeCountPredicate")
+				pv := obj.(*v1.PersistentVolume)
+				if pv.Spec.AWSElasticBlockStore != nil {
+					invalidPredicates.Insert("MaxEBSVolumeCount")
+				}
+				if pv.Spec.GCEPersistentDisk != nil {
+					invalidPredicates.Insert("MaxGCEPDVolumeCount")
+				}
+				if pv.Spec.AzureDisk != nil {
+					invalidPredicates.Insert("MaxAzureDiskVolumeCount")
+				}
+				c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(invalidPredicates)
+			},
+			DeleteFunc: func(obj interface{}) {
+				invalidPredicates := sets.NewString("MaxPDVolumeCountPredicate")
+				pv := obj.(*v1.PersistentVolume)
```

Comment:
This can be either a PersistentVolume or a cache.DeletedFinalStateUnknown object. See `deletePodFromCache` for example.